bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/directentrylogger/DirectEntryLogger.java

```diff
@@ -262,6 +263,7 @@ private ByteBuf internalReadEntry(long ledgerId, long entryId, long location, bo
             stats.getReadEntryStats().registerSuccessfulEvent(System.nanoTime() - start, TimeUnit.NANOSECONDS);
             return buf;
         } catch (EOFException eof) {
+            ReferenceCountUtil.release(buf);
```

Comment:
How about releasing it in the finally block? We can avoid memory leaks on some uncatched exceptions.

Reply:
Ah. Got it.

Reply:
Yes, if it can return buf correctly, then buf can be released by the upper layer